nucliadb/nucliadb/common/cluster/utils.py

```diff
@@ -27,23 +28,41 @@
 
 from .settings import settings
 
+_lock = asyncio.Lock()
+
+_STANDALONE_SERVER = "_standalone_service"
+
 
 async def setup_cluster() -> Union[KBShardManager, StandaloneKBShardManager]:
-    await setup_cluster_discovery()
-    mng: Union[KBShardManager, StandaloneKBShardManager]
-    if settings.standalone_mode:
-        mng = StandaloneKBShardManager()
-    else:
-        mng = KBShardManager()
-    set_utility(Utility.SHARD_MANAGER, mng)
-    return mng
+    async with _lock:
+        if get_utility(Utility.SHARD_MANAGER) is not None:
+            # already setup
+            return get_utility(Utility.SHARD_MANAGER)
+
+        await setup_cluster_discovery()
+        mng: Union[KBShardManager, StandaloneKBShardManager]
+        if settings.standalone_mode:
+            from .standalone.service import start_grpc
```

Comment:
what is the motivation for the lazy import? That module does not seem to have any global init